translator/translator.go

```diff
@@ -48,12 +52,16 @@ func ProtoMessageToDatastoreEntity(src proto.Message) (entity datastore.Entity,
 }
 
 // DatastoreEntityToProtoMessage converts any given datastore.Entity to supplied proto.Message
-func DatastoreEntityToProtoMessage(src datastore.Entity, dst proto.Message) (err error) {
+func DatastoreEntityToProtoMessage(src datastore.Entity, dst proto.Message, snakeCase bool) (err error) {
 	dstValues := reflect.ValueOf(dst).Elem()
 	for i := 0; i < dstValues.NumField(); i++ {
 		fName := dstValues.Type().Field(i).Name
 		if !strings.Contains(fName, "XXX_") {
-			fValue := GetProperty(src.Properties, fName)
+			keyName := fName
+			if snakeCase {
```

Comment:
We probably need a special case for enums where the enum keys should follow this format - ``FOO_BAR_BAZ`` - https://developers.google.com/protocol-buffers/docs/style.

Or would this work out of the box? How would Go lang struct field name look for enums?